server/src/main/java/org/apache/druid/server/coordinator/rules/PeriodLoadRule.java

```diff
@@ -41,16 +41,19 @@ public class PeriodLoadRule extends LoadRule
   private final Period period;
   private final boolean includeFuture;
   private final Map<String, Integer> tieredReplicants;
+  private final boolean useDefaultTierForNull;
 
   @JsonCreator
   public PeriodLoadRule(
       @JsonProperty("period") Period period,
       @JsonProperty("includeFuture") Boolean includeFuture,
-      @JsonProperty("tieredReplicants") Map<String, Integer> tieredReplicants
+      @JsonProperty("tieredReplicants") Map<String, Integer> tieredReplicants,
+      @JsonProperty("useDefaultTierForNull") @Nullable Boolean useDefaultTierForNull
   )
   {
-    this.tieredReplicants = tieredReplicants == null ? ImmutableMap.of(DruidServer.DEFAULT_TIER, DruidServer.DEFAULT_NUM_REPLICANTS) : tieredReplicants;
-    validateTieredReplicants(this.tieredReplicants);
+    this.useDefaultTierForNull = Configs.valueOrDefault(useDefaultTierForNull, true);
+    this.tieredReplicants = createTieredReplicants(tieredReplicants, this.useDefaultTierForNull);
+    validateTieredReplicants(this.tieredReplicants, this.useDefaultTierForNull);
```

Comment:
Since this is set via an API RulesResources#setDatasourceRules(). How would the exception thrown here be visible to end users. 

I guess we need to send a message back in the API response.

Reply:
In master, the message is returned as a warning after submitting the load rule. This behaviour here would be the same, or did I misunderstand what you mean?

<img width="583" alt="Screenshot 2023-06-20 at 8 39 57 AM" src="https://github.com/apache/druid/assets/9818861/fd844cf2-4638-4446-864b-221231087b39">

Reply:
Sure this looks good to me. We can share this screen shot as part of the PR description.